predicators/explorers/random_nsrts_explorer.py

```diff
@@ -0,0 +1,108 @@
+"""An explorer that takes random NSRTs."""
+
+from typing import Dict, List, Set
+
+from gym.spaces import Box
+
+from predicators import utils
+from predicators.explorers.base_explorer import BaseExplorer
+from predicators.structs import NSRT, Action, DummyOption, \
+    ExplorationStrategy, ParameterizedOption, Predicate, State, Task, Type, \
+    _GroundNSRT, _Option
+
+
+class RandomNSRTsExplorer(BaseExplorer):
+    """RandomNSRTsExplorer implementation.
+
+    Similar to RandomOptionsExplorer in that it chooses
+    uniformly at random out of whichever ground nsrts are
+    applicable in the current state, but different in that
+    the continuous parameter of the parameterized option is
+    generated by the nsrt's sampler rather than by sampling
+    uniformly at random from the option's parameter space.
+
+    This explorer is intended to be used when learning options
+    via reinforcement learning with oracle samplers. In this
+    setting, planning (e.g. using BilevelPlanningExplorer) is
+    not feasible because refinement is slow or fails when the
+    learned options aren't good enough yet.
+
+    Note that the sampler requires the current low-level state
+    as input. An alternative approach would generate a plan of
+    options by planning towards the goal with the ground nsrts
+    and avoid refinement by (1) inferring the terminal low-level
+    state of each option (the initial low-level state of the
+    subsequent option) from the option's sample, which is
+    proposing a low-level subgoal, and (2) setting the simulator
+    state directly to that state. Because applying dimensionality
+    reduction techniques on the sample makes this approach infeasible,
+    and this is something we may want to do, we avoid this approach
+    for now.
+    """
+
+    def __init__(self, predicates: Set[Predicate],
+                 options: Set[ParameterizedOption], types: Set[Type],
+                 action_space: Box, train_tasks: List[Task],
+                 nsrts: Set[NSRT]) -> None:
+
+        super().__init__(predicates, options, types, action_space, train_tasks)
+        self._nsrts = nsrts
+
+        # Sequence of options generated during exploration.
+        self._option_seq: Dict[int, List[_Option]] = {}
+
+    @classmethod
+    def get_name(cls) -> str:
+        return "random_nsrts"
+
+    def get_option_sequence(self, train_task_idx: int) -> List[_Option]:
+        """Get the sequence of options produced by the exploration strategy for
+        a particular task."""
+        return self._option_seq[train_task_idx]
+
+    def get_exploration_strategy(self, train_task_idx: int,
+                                 timeout: int) -> ExplorationStrategy:
+        # Ensure random over successive calls.
```

Comment:
is this comment left over from something? (I don't see what this is referring to)